nixos/modules/services/backup/tarsnap.nix

```diff
@@ -242,9 +254,11 @@ in
 
     systemd.services."tarsnap@" = {
       description = "Tarsnap archive '%i'";
-      requires    = [ "network.target" ];
+      requires    = [ "network-online.target" ];
+      after       = [ "network-online.target" ];
 
-      path = [ pkgs.tarsnap pkgs.coreutils ];
+      path = [ pkgs.tarsnap pkgs.coreutils pkgs.iputils ];
+      preStart = "while ! ping -q -c 1 betatest-server.tarsnap.com &> /dev/null; do true; done";
```

Comment:
I cannot check right now, but I believe tarsnap itself uses betatest-server.tarsnap.com to test network connectivity. I think the proposed patch basically replicates what tarsnap does anyway on network resolution failure. Perhaps @doshitan can elaborate on why it is necessary to do so.